### PR TITLE
Container/Column/ColumnSet/Improved visualizer

### PR DESCRIPTION
**Pull request by nesalang**
New UI Elements
- added container/column/columnSet

Improved Visualizer
- user can select json directly

Sample Cards Rendering
- Fixed most of style issues when rendered with sample cards

### CODE REVIEW

File: source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRColumnRenderer.mm [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+#import "Column.h"
+#import "AdaptiveCard.h"
+
+@implementation ACRColumnRenderer
+
++ (ACRColumnRenderer* ) getInstance
+{
+    static ACRColumnRenderer *singletonInstance = nil;
+    static dispatch_once_t predicate;
+    dispatch_once(&predicate, ^{singletonInstance = [[self alloc] init];});
```
Comment (bviglietta) [inline source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRColumnRenderer.mm]: There's a shorter way to say this:
`static ARColumnRenderer *singletonInstance = [[self alloc] init];`

File: source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRBaseCardElementRenderer.mm [hunk 2]
```diff
@@ -0,0 +1,31 @@
+//
+//  ACRBaseCardElementRenderer
+//  ACRBaseCardElementRenderer.h
```
Comment (bviglietta) [inline source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRBaseCardElementRenderer.mm]: >h [](start = 31, length = 1)

mm

File: source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRBaseCardElementRenderer.mm [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+//
+//  Copyright © 2017 Microsoft. All rights reserved.
+//
+
+#import "ACRBaseCardElementRenderer.h"
+
+@implementation ACRBaseCardElementRenderer
```
Comment (bviglietta) [inline source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRBaseCardElementRenderer.mm]: >@implementation [](start = 0, length = 15)

Three methods in this file and three different bracing styles. Please pick a consistent style (preferably one shared by the whole project).

File: source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRColumnStackView.mm [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    CGSize test;
+    if(self.relativeColumnWidth)
+        return test = CGSizeMake(self.relativeColumnWidth*3, 150);
+    else
+    test = super.intrinsicContentSize;
+    test.width = test.width * self.relativeColumnWidth;
+    return test;
+}
+#endif
+
```
Comment (bviglietta) [inline source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRColumnStackView.mm]: Is this test code? If so, it should be removed. If not, it should be cleaned up.

File: source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRRegistration.mm [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+}
+
+- (UIView* ) render: (UIView*) view
+    withContentView: (UIStackView*) newView
+      withCardElems: (std::vector<std::shared_ptr<BaseCardElement>> const &) elems
+      andHostConfig: (std::shared_ptr<HostConfig> const &) config
+{ 
+    
+    if([view class] == [UIStackView class]){
```
Comment (bviglietta) [inline source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRRegistration.mm]: >[view class] == [UIStackView class] [](start = 7, length = 35)

Prefer `[view isKindOfClass:[UIStackView class]]`. That will work even if the view is a subclass of UIStackView.

File: source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRImageRenderer.mm [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+//  Copyright © 2017 Microsoft. All rights reserved.
+//
+
+#import "ACRImageRenderer.h"
+#import "Image.h"
+#import "SharedAdaptiveCard.h"
+
+@implementation ACRImageRenderer
+
++ (ACRImageRenderer* ) getInstance
```
Comment (riarenas) [inline source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRImageRenderer.mm]: >  [](start = 20, length = 1)

[nit] extra space

Comment (riarenas) [inline source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRImageRenderer.mm]: There's a few of these

---
In reply to: [125322115](https://github.com/Microsoft/AdaptiveCards/pull/451#discussion_r125322115) [](ancestors = 125322115)

File: source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRTextBlockRenderer.mm [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+- (UIColor* ) getTextBlockColor:(std::shared_ptr<TextBlock> const &) txtBlock
+                 withHostConfig:(std::shared_ptr<HostConfig> const &) config
+{
+    long num = 0;
+    std::string str;
+    switch (txtBlock->GetTextColor()) {
+        case TextColor::Dark: {
+            str = (txtBlock->GetIsSubtle())?
+                config->colors.dark.subtle: config->colors.dark.normal;
```
Comment (riarenas) [inline source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRTextBlockRenderer.mm]: [nit] add a space before ? and :  when you use this operator for readability.

File: source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRImageRenderer.mm [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+            return UIViewContentModeLeft;
+    }
+}
+- (UIView* ) render: (UIStackView*) viewGroup
+       withCardElem: (std::shared_ptr<BaseCardElement> const &) elem
+      andHostConfig: (std::shared_ptr<HostConfig> const &) config
+{
+    std::shared_ptr<Image> imgElem = std::dynamic_pointer_cast<Image>(elem);
+
+    NSURL *url =
```
Comment (riarenas) [inline source/ios/AdaptiveCards/ADCIOSFramework/ADCIOSFramework/ACRImageRenderer.mm]: >NSURL *url  [](start = 4, length = 11)

[nit] NSURL* url

**Pull request merged**